client/client.go

```diff
@@ -0,0 +1,166 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+
+	"github.com/coredns/coredns/pb"
+	"github.com/coredns/coredns/plugin/pkg/tls"
+
+	"github.com/miekg/dns"
+	"google.golang.org/grpc"
+	creds "google.golang.org/grpc/credentials"
+)
+
+// Client provides a convenient interface to a gRPC-based DNS service.
+type Client struct {
+	pbClient pb.DnsServiceClient
+}
+
+// Watch is used to track access results from watching a particular query.
+type Watch struct {
+	WatchID int64
+	Msgs    chan *dns.Msg
+	stream  pb.DnsService_WatchClient
+	client  *Client
+}
+
+// NewClient establishes a connection to a server and returns a pointer to a Client.
+func NewClient(endpoint, cert, key, ca string) (*Client, error) {
```

Comment:
We should add a `[]grpc.DialOption` parameter that we append all our options to. This way they can set backoff policies, etc.